objects.py

```diff
@@ -5,40 +5,50 @@
 
 
 class Dessert(object):
-
-    def __init__():
+    def __init__(self,price,calories=None):
         # Edit me!
         # You need to be able to initialize a Dessert object with arguments:
         # price - required
         # calories - optional
-
+        self.price = price
+        self.calories = calories
         # This should set the object's price and calories, accessible by
         # .price and .calories respectively.
-        pass
 
     # Add a calories_per_dollar method that returns the calories per dollar
     # for the dessert.
-
+    def calories_per_dollar(self):
+        if self.calories == None:
+            return None
+        else:
+            ratio = self.calories/self.price
+            return ratio
     # Define a method is_a_cake on Dessert that returns False
+    def is_a_cake(self):
+        return False
+
 
 
-class Cake(Dessert):
 
-    def __init__():
+class Cake(Dessert):
+    def __init__(self,kind):
         # Edit me!
         # Cakes all cost the same amount and have the same calories, so their
         # price and calories can be set at the class-level, not during init.
         # However, we need to be able to tell cakes apart. Accept argument:
         # kind - required
-
-        pass
-
+        self.kind = kind
+        price = 5
+        calories = 200
+        super(Cake,self).__init__(price,calories)
```

Comment:
you don't need to do this, although it works! initializing your parent is mostly done if you really need to call whatever logic is inside its `__init__`. if it's just a case of creating constant values, you can define them at the class level:

```
class Cake(Dessert):
    price = 5
    calories = 200

   def __init__(self, kind):
     ... etc ...
```

then `self.price` will always be 5 for a `Cake`, and so on.